core/domain/caching_services.py

```diff
@@ -80,7 +87,12 @@
 # and Hashes. More details can be found at: https://redis.io/topics/data-types.
 CACHE_NAMESPACE_DEFAULT = 'default'
 
-DESERIALIZATION_FUNCTIONS = {
+# Dict DESERIALIZATION_FUNCTIONS has key-value pair of string and
+# functions respectively. Every function accept one argument which is string
+# and returns corresponding object, which can be any of the type Collection,
+# Exploration, Skill, Story, Topic, PlatformParameter and simple python object.
+# So, Any was assigned as a type for return value of function.
+DESERIALIZATION_FUNCTIONS: Dict[str, Callable[[str], Any]] = {
```

Comment:
Thinking about this, can we use `TypedDict` here? That way we don't need ti use `Any`.

Reply:
if we use `typeDict` like 
![Screenshot from 2022-01-06 16-36-08](https://user-images.githubusercontent.com/12553576/148374248-99d764c9-46f5-4403-9393-b7e6cb7a5ab1.png)
then it gives following error :
![Screenshot from 2022-01-06 16-36-15](https://user-images.githubusercontent.com/12553576/148374351-88fdf55a-2c39-4611-a632-957cc93fb8e3.png)

expected reason for above problem:
if we write `CACHE_NAMESPACE_COLLECTION` in `TypeDict` then typedict expect it to be a key with string "CACHE_NAMESPACE_COLLECTION"

or if we use `typeDict` in following manner:
![Screenshot from 2022-01-06 16-36-34](https://user-images.githubusercontent.com/12553576/148374448-2444856e-25ea-4aec-86b2-b103b04c3591.png)
then it gives following errors:
![Screenshot from 2022-01-06 16-37-07](https://user-images.githubusercontent.com/12553576/148374525-ab5b566d-9e65-425b-93de-3d7a4972ab56.png)

expected reason for above problem:
here, `TypeDict` needs direct string literals but in `DESERIALIZATION_FUNCTIONS` all keys are mapped with variables.
for eg:- `CACHE_NAMESPACE_COLLECTION` is key which have value `collection`(string).

Reply:
I think you can declare the constants as literals https://mypy.readthedocs.io/en/stable/literal_types.html